controllers/space/space_controller.go

```diff
@@ -269,6 +296,41 @@ func (r *Reconciler) newNSTemplateSet(namespace string, name string, tmplTier *t
 	return nsTmplSet
 }
 
+func (r *Reconciler) newNSTemplateSetSpec(space *toolchainv1alpha1.Space, tmplTier *toolchainv1alpha1.NSTemplateTier) toolchainv1alpha1.NSTemplateSetSpec {
+	s := toolchainv1alpha1.NSTemplateSetSpec{
+		TierName: space.Spec.TierName,
+	}
+	if tmplTier.Spec.ClusterResources != nil {
+		s.ClusterResources = &toolchainv1alpha1.NSTemplateSetClusterResources{
+			TemplateRef: tmplTier.Spec.ClusterResources.TemplateRef,
+		}
+	}
+	if len(tmplTier.Spec.Namespaces) > 0 {
+		s.Namespaces = make([]toolchainv1alpha1.NSTemplateSetNamespace, len(tmplTier.Spec.Namespaces))
+		for i, ns := range tmplTier.Spec.Namespaces {
+			s.Namespaces[i] = toolchainv1alpha1.NSTemplateSetNamespace(ns)
+		}
+	}
+	// space roles
+	if len(tmplTier.Spec.SpaceRoles) > 0 {
+		s.SpaceRoles = make([]toolchainv1alpha1.NSTemplateSetSpaceRole, len(tmplTier.Spec.SpaceRoles))
+		// append by alphabetical order of role names
+		roles := make([]string, 0, len(tmplTier.Spec.SpaceRoles))
+		for r := range tmplTier.Spec.SpaceRoles {
+			roles = append(roles, r)
+		}
+		sort.Strings(roles)
+		for i, r := range roles {
+			sr := tmplTier.Spec.SpaceRoles[r]
+			s.SpaceRoles[i] = toolchainv1alpha1.NSTemplateSetSpaceRole{
+				TemplateRef: sr.TemplateRef,
+				// TODO: include usernames from SpaceBindings
+			}
+		}
```

Comment:
The SpaceRoles should be set only when there is a SpaceBinding set - otherwise, it should be empty. There is no reason for adding the SpaceRole TierTemplate refs to the NSTemplate if there is no SpaceBinding for the role set.

Reply:
done in 4d0f300